assets/src/hooks/use_api_response.tsx

```diff
@@ -2,24 +2,39 @@ import { useEffect, useState } from "react";
 import { isDup } from "Util/util";
 import useInterval from "Hooks/use_interval";
 import { useLocation } from "react-router-dom";
+import * as Sentry from "@sentry/react";
 
 const MINUTE_IN_MS = 60_000;
 
-const FAILURE_RESPONSE = { success: false };
+const FAILURE_RESPONSE = { success: false, lastSuccess: null };
 
 const doFailureBuffer = (
-  lastSuccess: number,
+  lastSuccess: number | null,
   failureModeElapsedMs: number,
   setApiResponse: React.Dispatch<React.SetStateAction<object>>,
+  setLastResponseWasFailure: React.Dispatch<React.SetStateAction<boolean>>,
   apiResponse: object = FAILURE_RESPONSE
 ) => {
-  const elapsedMs = Date.now() - lastSuccess;
-
-  if (elapsedMs < failureModeElapsedMs) {
+  if (lastSuccess == null) {
+    // We haven't had a successful request since initial page load.
+    // Continue showing the initial "no data" state.
     setApiResponse((state) => state);
-  }
-  if (elapsedMs >= failureModeElapsedMs) {
-    setApiResponse(apiResponse);
+  } else {
+    const elapsedMs = Date.now() - lastSuccess;
+
+    // Because we recycle state when a failure occurs,
+    // we need to track failures in a separate state variable.
+    setLastResponseWasFailure(true);
+
+    if (elapsedMs < failureModeElapsedMs) {
+      setApiResponse((state) => state);
+    }
+    if (elapsedMs >= failureModeElapsedMs) {
+      setApiResponse(apiResponse);
+    }
+
+    // This will trigger until a success API response is received.
+    Sentry.captureMessage("API response failure encountered.");
```

Comment:
I _think_ that makes sense. Don't we want to know when a screen gets a failure response (as long as it isn't the failure response on page load)? This function is only called when a failure happens so that seems like the perfect place for the log.

Reply:
Based on convo in other thread, got this changed to just log when entering `NoData` state.